edu.cuny.hunter.streamrefactoring.core/META-INF/MANIFEST.MF

```diff
@@ -22,24 +18,18 @@ Require-Bundle: org.eclipse.ltk.core.refactoring,
  com.ibm.wala.shrike;bundle-version="1.3.10",
  com.ibm.safe.core;bundle-version="1.0.0",
  com.ibm.safe.typestate;bundle-version="1.0.0",
- org.eclipse.jdt.launching
+ org.eclipse.jdt.launching,
+ org.eclipse.core.resources,
+ org.eclipse.core.runtime,
+ org.eclipse.jdt.core.manipulation,
+ org.eclipse.text,
+ org.eclipse.ui,
+ org.eclipse.pde.core,
+ org.objenesis
 Export-Package: edu.cuny.hunter.streamrefactoring.core.analysis,
  edu.cuny.hunter.streamrefactoring.core.contributions,
  edu.cuny.hunter.streamrefactoring.core.descriptors,
  edu.cuny.hunter.streamrefactoring.core.messages,
  edu.cuny.hunter.streamrefactoring.core.refactorings,
  edu.cuny.hunter.streamrefactoring.core.utils,
  edu.cuny.hunter.streamrefactoring.core.wala
-Import-Package: com.ibm.safe.controller,
```

Comment:
What's wrong with `Import-Package`?

Reply:
"Import-Package" means that this particular package should exist in the platform, but you are not specifying the plugin that provides it. 
So it is a weakly declared dependency, in the sense you cannot actually import the package unless it is present in your platform (i.e. it has been installed). 
Moving this dependency declaration to a "Require-Bundle" means that maven and eclipse p2 both understand the dependency and will deal with it appropriately (i.e. by deploying the plugin you depend on, and its own dependencies). So for a "managed" build it is recommended not to use "Import-Package", I removed mostly all of those.

The problem is then to find a "bundle" or plugin that provides the required package. We can resolve such artifacts in the update sites we provide to maven :  https://github.com/ponder-lab/Optimize-Java-8-Streams-Refactoring/pull/209/files/b7f6acf331a63a36a04345feff1837490ef36337#diff-b45a7fd496ea4028cf723e4f5ecfcd38R108-R153

In this place I put the WALA and SAFE repositories, as well as point "orbit" which is an eclipse project providing quite a few common utilities packaged as bundles so we can use them in this kind of integration scenario. e.g. this is where we resolve dependencies on apache stuff https://download.eclipse.org/tools/orbit/downloads/drops/R20200529191137/ 
For instance this is where we now resolve org.apache.commons.csv dependencies (instead of packaging them ourselves in .eval plugin)

Reply:
Oh, interesting. I didn't know about that. Thanks!